src/main/java/io/github/acm19/aws/interceptor/http/RequestSigner.java

```diff
@@ -49,7 +48,7 @@ class RequestSigner {
      * @param region
      */
     RequestSigner(String service,
-                  Signer signer,
+                  AwsV4HttpSigner signer,
```

Comment:
It seems more sensible to use the base interface instead `software.amazon.awssdk.http.auth.spi.signer.HttpSigner`. So if we change the implementation to a different signer it's transparent to the users.

Reply:
agree - i did attempt it before but couldn't get it to work because I forgot how java works